build.sbt

```diff
@@ -5126,12 +5125,41 @@ buildEngineDistribution := {
   log.info(s"Engine package created at $root")
 }
 
+lazy val buildEngineDistribution =
+  taskKey[Unit]("Builds the engine distribution and optionally native image")
+buildEngineDistribution := Def.taskIf {
+  if (shouldBuildNativeImage.value) {
+    createEnginePackage.value
+    (`engine-runner` / buildNativeImage).value
+  } else {
+    createEnginePackage.value
+  }
+}.value
+
 // This makes the buildEngineDistribution task usable as a dependency
 // of other tasks.
 ThisBuild / buildEngineDistribution := {
   buildEngineDistribution.result.value
 }
 
+lazy val shouldBuildNativeImage = taskKey[Boolean](
+  "Whether native image should be build within buildEngineDistribution task"
+)
+
+ThisBuild / shouldBuildNativeImage := {
+  val prop = System.getenv("ENSO_LAUNCHER")
+  BuildInfo.isReleaseMode || prop == "native" || prop == "debugnative"
```

Comment:
Right now we shouldn't be building _native image_ launchers unless `ENSO_LAUNCHER=native` or `ENSO_LAUNCHER=debugnative`. Why are we checking `BuildInfo.isReleaseMode` here?

Reply:
Yes, we shouldn't build it for release.

Reply:
Good point. Removed this check in d958a248376c5e4c366d3bd1dbfec9048f368589